lib/net.js

```diff
@@ -1300,6 +1300,11 @@ function listen(self, address, port, addressType, backlog, fd, exclusive) {
 Server.prototype.listen = function() {
   var self = this;
 
+  if (self._listenHasBeenCalled) 
+    throw new Error('cannot call "listen" multiple times');
+
+  self._listenHasBeenCalled = true;
```

Comment:
Wondering, if it's better to initialize server with `self._listenHasBeenCalled = false` to avoid hidden class changes? Or am I wrong?

/cc @thefourtheye